### PR TITLE
feat(compute): add compute reservation samples (get, list and create with template)

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description
Implemented code for three samples - compute_reservation_get , compute_reservation_list 
 and compute_reservation_create_template. To be able to create reservation for Instance Template with regional location in the ReservationIT class are implemented methods to create and delete such Instance Template. Additional task was create for such implementation and ReservationIT class will be fixed after this merge. 

In the HyperdiskIT class two tests were disabled due the errors caused by daily limit for creating Storage Pool. They will be enabled after approval.
One sample was done for Python -
1.  https://github.com/GoogleCloudPlatform/python-docs-samples/blob/3d6340881961a238b2ecc3a0028a02aa2ee9d944/compute/client_library/snippets/instance_templates/create_reservation_from_template.py#L22

These samples were done for Goland -
1. https://github.com/GoogleCloudPlatform/golang-samples/blob/569f34aff36fe05a59880cc49f8ca7436cc6d19a/compute/reservations/create_reservation.go
2. https://github.com/GoogleCloudPlatform/golang-samples/blob/569f34aff36fe05a59880cc49f8ca7436cc6d19a/compute/reservations/get_reservation.go
3. https://github.com/GoogleCloudPlatform/golang-samples/blob/569f34aff36fe05a59880cc49f8ca7436cc6d19a/compute/reservations/list_reservations.go

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/reservation/CreateReservationForInstanceTemplate.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+    // Name of the zone in which you want to create the reservation.
+    String zone = "us-central1-a";
+    // Name of the reservation you want to create.
+    String reservationName = "YOUR_RESERVATION_NAME";
+    // The number of virtual machines you want to create.
+    int numberOfVms = 3;
+    // The URI of the instance template with GLOBAL Location
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/reservation/CreateReservationForInstanceTemplate.java:43]:
```suggestion
    // The URI of the instance template with GLOBAL location
```

File: compute/cloud-client/src/main/java/compute/reservation/CreateReservationForInstanceTemplate.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    // to be used for creating the reservation.
+    String instanceTemplateUri =
+        "projects/YOUR_PROJECT_ID/global/instanceTemplates/YOUR_INSTANCE_TEMPLATE_NAME";
+
+    // The URI of the instance template with REGIONAL Location
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/reservation/CreateReservationForInstanceTemplate.java:48]:
```suggestion
    // The URI of the instance template with REGIONAL location
```

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/test/java/compute/reservation/ReservationIT.java
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/reservation/ReservationIT.java:1]: Thank you

File: compute/cloud-client/src/test/java/compute/reservation/ReservationIT.java [hunk 3]
```diff
@@ -116,4 +180,150 @@ public void testCreateReservation()
...
+  }
+
+  @Test
+  public void firstCrateReservationWithGlobalInstanceTemplateTest()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    CreateReservationForInstanceTemplate.createReservationForInstanceTemplate(
+        PROJECT_ID, RESERVATION_NAME_GLOBAL,
+        GLOBAL_INSTANCE_TEMPLATE_URI, NUMBER_OF_VMS, ZONE);
+
+    try (ReservationsClient reservationsClient = ReservationsClient.create()) {
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/reservation/ReservationIT.java:211]: Tried to do it in this way, but got errors. Didn't see such implementation for the other classes. Could we left it as is?

**Review by Sita04 — APPROVED** [review]

**Pull request merged**